src/include/OSL/oslconfig.h.in

```diff
@@ -225,3 +225,29 @@ using ConstIndex = std::integral_constant<int, N>;
 
 
 OSL_NAMESPACE_EXIT
+
+
+namespace std {  // not necessary in C++17, then we can just say std::hash
+#ifndef OIIO_USTRING_HAS_STDHASH
+// Not a new enough OIIO to define std::hash<ustring>, so we'll do it
+template<> struct hash<OSL::ustring> {
+    std::size_t operator()(OSL::ustring u) const noexcept { return u.hash(); }
```

Comment:
Can this be made constexpr?

Reply:
no, the hash is stored in dynamic memory somewhere in the internal table